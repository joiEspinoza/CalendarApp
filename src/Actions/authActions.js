import Swal from 'sweetalert2';
import { fetchConToken, fetchSinToken } from '../Helpers/fetch';
import { types } from '../Type/types';

//////

const startLogin = ( email, password ) =>
{
    return async ( dispatch ) => 
    {
        try 
        {

            const response = await fetchSinToken( "auth", { email, password }, "POST" );
            const body = await response.json();

            if( body.ok )
            {
                localStorage.setItem( "token", body.jsonWebToken );
                localStorage.setItem( "creacionToken" , new Date().getTime() );

                dispatch( login( { uid : body.uid, name : body.name } ) );
            }
            else
            {

                if( !body.errors )
                {
                    return Swal.fire( "error", body.msg, "error" );
                };

                if( body.errors.password )
                {
                    return Swal.fire( "error", body.errors.password.msg, "error" );
                };

                if( body.errors.email )
                {
                    return Swal.fire( "error", body.errors.email.msg, "error" );
                };

            };

        } 
        catch( error ) 
        {
            console.log( error );
            return Swal.fire( "error", error, "error" );
        };
    };
}

const login = ( user ) => ( { type : types.authLogin, payload : user } );


const starRegister = ( name, email, password ) =>
{
    
    return async ( dispatch ) => 
    {
       try 
       {

            const response = await fetchSinToken( "auth/register", { name, email, password }, "POST" );
            const data = await response.json();

            if( data.ok )
            {
                
                localStorage.setItem( "token", data.jsonWebToken );
                localStorage.setItem( "creacionToken" , new Date().getTime() );

                dispatch( login( { uid : data.uid, name : data.name } ) );

                Swal.fire( "Listo", "Usuario creado exitosamente", "success" );
                
            }
            else
            {
        

                if( !data.errors )
                {
                    return Swal.fire( "error", data.msg, "error" );
                };
                
                if( data.errors.name )
                {
                    return Swal.fire( "error", data.errors.name.msg, "error" );
                }

                if( data.errors.email )
                {
                    return Swal.fire( "error", data.errors.email.msg, "error" );
                }

                if( data.errors.password )
                {
                    return Swal.fire( "error", data.errors.password.msg, "error" );
                }

            
            };
       } 
       catch( error ) 
       {
            console.log( error );
            return Swal.fire( "error", error,"error" );
       };

    };
};


const startChecking = () =>
{
    return async ( dispatch ) => 
    {

        try 
        {
            
            const response = await fetchConToken( "auth/revalidtoken" );
            const data = await response.json();
    
    
            if( data.ok )
            {
                    
                localStorage.setItem( "token", data.jsonWebToken );
                localStorage.setItem( "creacionToken" , new Date().getTime() );
    
                dispatch( login( { uid : data.uid, name : data.name } ) );
                    
            }
            else
            {
                dispatch( checkingFinish() );
            };

        } 
        catch( error ) 
        {
            
        }

    };
};


const checkingFinish = () => ( { type : types.authCheckingFinish } );


const startLogout = () =>
{

    return ( dispatch ) =>
    {
        localStorage.clear();
        dispatch( logout() );
    };

};

const logout = () => ( { type : types.authLogout } );


////

export { startLogin, starRegister, startChecking, startLogout , login  };