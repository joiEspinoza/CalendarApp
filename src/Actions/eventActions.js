import Swal from "sweetalert2";
import { fetchConToken } from "../Helpers/fetch";
import { prepararEvento } from "../Helpers/prepararEvento";
import { types } from "../Type/types";

//////

const eventStartAddNew = ( evento ) =>
{
    return async ( dispatch, getState ) =>
    {
        try 
        {

            const { uid, name } = getState().auth;

            const response =  await fetchConToken( "events", evento, "POST" );

            const data = await response.json();

            if( data.ok )
            {
                
                evento.id = data.evento.id;
                evento.user = { _id : uid, name };
                console.log( evento );
                dispatch( eventAddNewAction( evento )  );
            };


        } 
        catch( error ) 
        {
                console.log( error );
        };

    };
};

const eventAddNewAction = ( event ) => ( { type : types.eventAddNew, payload : event } );

const eventStartLoading = () =>
{
    return async ( dispatch ) =>
    {

        try 
        {
            const response = await fetchConToken( "events" );
            const data = await response.json();
    
            const eventos =  prepararEvento( data.eventos );

            dispatch( eventLoading( eventos ) );

        } 
        catch( error ) 
        {
            console.log( error );
        };


    };
};

const eventLoading = ( eventos ) => ( { type : types.eventLoad, payload : eventos } );

const startEventUpdateAction = ( updateEvent ) =>
{
    return async ( dispatch, getState ) => 
    {

        const { id } = getState().calendar.activeEvents;

        try 
        {
            const response =  await fetchConToken( `events/${ id }`, updateEvent , "PUT" ); 
            
            const data = await response.json();

            if( data.ok )
            {
                dispatch( eventUpdateAction( data.evento ) );
            }
            else
            {
                Swal.fire( "error",data.msg,"error" );
            };

        } 
        catch( error ) 
        { 
            console.log( error );
        };

    };
};

const eventUpdateAction = ( event ) => ( { type : types.eventUpdate, payload : event } );


const startDeleteAction = () =>
{
    return  async ( dispatch, getState ) =>
    {
        try 
        {
            const { id } = getState().calendar.activeEvents;

            const response = await fetchConToken( `events/${ id }`, {} , "DELETE" );

            const data = await response.json();

            if( data.ok )
            {
                dispatch( eventDeleteAction() );
            }
            else
            {
                Swal.fire( "error", data.msg, "error" );
            };


        } 
        catch( error ) 
        {
            console.log( error );
        };
    };
};

const eventDeleteAction = () => ( { type : types.eventDelete } );

//////////////////////////

const eventLogout = () => ( { type : types.eventLogout } );

const eventSetActiveAction = ( event ) => ( { type : types.eventSetActive, payload : event } );

const eventClearActiveEventAction = () => ( { type : types.eventClearActive } );


/////

export 
{  
    eventStartAddNew,
    eventStartLoading,
    startDeleteAction,
    startEventUpdateAction,
    eventSetActiveAction, 
    eventClearActiveEventAction,
    eventLogout
};