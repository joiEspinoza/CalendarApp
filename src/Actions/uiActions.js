const { types } = require("../Type/types");

/////

const uiOpenModalAction = () => ( { type : types.uiOpenModal } );

const uiCloseModalAction = () => ( { type : types.uiCloseModal } );

/////

export { uiOpenModalAction, uiCloseModalAction };