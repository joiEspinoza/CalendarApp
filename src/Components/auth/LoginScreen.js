import React from 'react';
import { useDispatch } from 'react-redux';
import Swal from 'sweetalert2';
import { starRegister, startLogin } from '../../Actions/authActions';
import useForm from '../../hooks/useForm';
import "../../Style/login.css"

/////

const LoginScreen = () => 
{

    const dispatch = useDispatch();

    //////

    const initLoginFormValues = 
    {
        correoIngreso : "king@gmail.com",
        passwordIngreso : "123456",
    };

    const [ formLoginValues, handleLoginInputChange ] = useForm( initLoginFormValues );

    const { correoIngreso, passwordIngreso } = formLoginValues;

    const handleLogin = ( e ) =>
    {
        e.preventDefault();
        dispatch( startLogin( correoIngreso, passwordIngreso  ) );

    };

    /////////

    const initRegisterFormValues = 
    {
        nombreRegistro : "",
        correoRegistro : "",
        passwordRegistro : "",
        passwordRegistro2 : "",
    };

    const [ formRegisterValues, handleRegisterInputChange ] = useForm( initRegisterFormValues );

    const { nombreRegistro, correoRegistro, passwordRegistro, passwordRegistro2 } = formRegisterValues;

    const handleRegister = ( e ) =>
    {
        e.preventDefault();
        
        if( passwordRegistro !== passwordRegistro2 )
        {
            return Swal.fire( "error", "las contrase単as deben ser iguales", "error" );
        };

        dispatch( starRegister( nombreRegistro, correoRegistro, passwordRegistro ) );

    };

    //////////

  

    ///////////////////////////////////////////////////////


    return (

            <div className="container login-container">

                <div className="row">

                    <div className="col-md-6 login-form-1">

                        <h3>Ingreso</h3>

                        <form onSubmit={ handleLogin }>

                            <div className="form-group">
                                <input type="email" className="form-control" placeholder="Correo" name="correoIngreso" value={ correoIngreso } onChange={ handleLoginInputChange } />
                            </div>

                            <div className="form-group">
                                <input type="password" className="form-control" placeholder="Contrase単a" name="passwordIngreso" value={ passwordIngreso } onChange={ handleLoginInputChange } />
                            </div>

                            <div className="form-group">
                                <input type="submit" className="btnSubmit" value="Login" />
                            </div>

                        </form>

                    </div>
    
                    <div className="col-md-6 login-form-2">

                        <h3>Registro</h3>

                        <form onSubmit={ handleRegister } >

                            <div className="form-group">
                                <input type="text"className="form-control" placeholder="Nombre" name="nombreRegistro" value={ nombreRegistro } onChange={ handleRegisterInputChange } />
                            </div>

                            <div className="form-group">
                                 <input type="email" className="form-control" placeholder="Correo" name="correoRegistro" value={ correoRegistro } onChange={ handleRegisterInputChange } />
                            </div>

                            <div className="form-group">
                                <input type="password" className="form-control" placeholder="Contrase単a" name="passwordRegistro" value={ passwordRegistro } onChange={ handleRegisterInputChange } />
                            </div>
    
                            <div className="form-group">
                                <input type="password" className="form-control" placeholder="Repita la contrase単a" name="passwordRegistro2" value={ passwordRegistro2 } onChange={ handleRegisterInputChange } />
                            </div>
    
                            <div className="form-group">
                                <input type="submit" className="btnSubmit"  value="Crear cuenta" />
                            </div>

                        </form>

                    </div>

                </div>

            </div>
        );

};

///////

export default LoginScreen;
