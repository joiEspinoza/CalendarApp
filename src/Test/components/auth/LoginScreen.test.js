import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from "redux-thunk";
import "@testing-library/jest-dom";
import LoginScreen from '../../../Components/auth/LoginScreen';
import { starRegister, startLogin } from '../../../Actions/authActions';
import Swal from 'sweetalert2';

/////

jest.mock( '../../../Actions/authActions', () =>
( 
    { 
        startLogin : jest.fn(),
        starRegister : jest.fn()
    } 

));


////

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

const initState = {};
const store = mockStore( initState );
store.dispatch = jest.fn();

//////

describe('Pruebas en LoginScreen.js', () => 
{
    
    beforeEach( () =>
    {

        jest.clearAllMocks();

    } );

    ////

    const wrapper = mount( 
    
        <Provider store={ store }>
            <LoginScreen/> 
        </Provider>
    
    );
   
    /////

    test('Debe mostrarse correctamente ', () => 
    {
        expect( wrapper ).toMatchSnapshot();
    });


    test('Debe llamar el dispatch del login', () => 
    {
        
        wrapper.find( 'input[name="correoIngreso"]' ).simulate( "change", 
        {

            target : { name : "correoIngreso", value : "king2@gmail.com" }

        } );

        wrapper.find( 'input[name="passwordIngreso"]' ).simulate( "change", 
        {

            target : { name : "passwordIngreso", value : "123456" }

        } );

        wrapper.find( "form" ).at( 0 ).prop( "onSubmit" )( {  

            preventDefault(){}

        } );

        expect( startLogin ).toHaveBeenCalledWith( "king2@gmail.com", "123456" );

    });



    test('No hay registro si las contraseñas no son iguales', () => 
    {
      
        wrapper.find( 'input[name="passwordRegistro"]' ).simulate( "change", 
        {

            target : { name : "passwordRegistro", value : "ABCDE" }

        } );


        wrapper.find( 'input[name="passwordRegistro2"]' ).simulate( "change", 
        {

            target : { name : "passwordRegistro2", value : "1234ABCDE" }

        } );

        wrapper.find( "form" ).at( 1 ).prop( "onSubmit" )( {  

            preventDefault(){}

        } );

        expect( starRegister ).toHaveBeenCalledTimes( 0 );
        expect( starRegister ).not.toHaveBeenCalled();
        expect( Swal.fire ).toHaveBeenCalledWith( "error", "las contraseñas deben ser iguales", "error" );


    });

    test('Debe disparar el registro con contraseñas iguales', () => 
    {
        
        wrapper.find( 'input[name="passwordRegistro"]' ).simulate( "change", 
        {

            target : { name : "passwordRegistro", value : "ABCDE" }

        } );


        wrapper.find( 'input[name="passwordRegistro2"]' ).simulate( "change", 
        {

            target : { name : "passwordRegistro2", value : "ABCDE" }

        } );

        wrapper.find( "form" ).at( 1 ).prop( "onSubmit" )( {  

            preventDefault(){}

        } );

        expect( starRegister ).toHaveBeenCalledTimes( 1 );

    });
    
    
    
});
