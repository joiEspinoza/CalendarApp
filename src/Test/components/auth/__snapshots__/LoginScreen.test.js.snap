// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en LoginScreen.js Debe mostrarse correctamente  1`] = `
<div
  className="container login-container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 login-form-1"
    >
      <h3>
        Ingreso
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="correoIngreso"
            onChange={[Function]}
            placeholder="Correo"
            type="email"
            value="king@gmail.com"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="passwordIngreso"
            onChange={[Function]}
            placeholder="Contraseña"
            type="password"
            value="123456"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="btnSubmit"
            type="submit"
            value="Login"
          />
        </div>
      </form>
    </div>
    <div
      className="col-md-6 login-form-2"
    >
      <h3>
        Registro
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="nombreRegistro"
            onChange={[Function]}
            placeholder="Nombre"
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="correoRegistro"
            onChange={[Function]}
            placeholder="Correo"
            type="email"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="passwordRegistro"
            onChange={[Function]}
            placeholder="Contraseña"
            type="password"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="passwordRegistro2"
            onChange={[Function]}
            placeholder="Repita la contraseña"
            type="password"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="btnSubmit"
            type="submit"
            value="Crear cuenta"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
